### ISSUE TITLE
Custom BEM CSS classes not generated - randomly

### ISSUE THREAD

**Issue by xApep** [tailwindlabs/tailwindcss#14198]
hey!

I have base.css with code:
```
@layer base {
  .cards {
    @apply px-4 md:mx-auto md:px-10 xl:px-16 xl:max-w-[1536px];

    &--mobile-none {
      @apply -mx-4 md:mx-0;
    }
  }
}
```

which i import it in tailwind.css 
```
@import 'tailwindcss/base';
@import './base.css';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
```

my postcss.config.js
```
module.exports = {
  plugins: {
    'postcss-import': {},
    'tailwindcss/nesting': {},
    tailwindcss: {},
    autoprefixer: {},
  },
}
```

Now the problem is that when I write some "custom" css components it happens that very randomly some classes, using BEM methodology it is just not generated.

Few things to have in mind.
- those css classes are set in CMS systems, so they never exist for tailwind content watch; but since classes are set in .css they should always be generated?
- I have tried with or without adding *.css to "content" in tailwind settings, but problems are more or less always the same

The solution is, either I define some plain .html file, which tailwind "content" can watch, or if I define classes without BEM, so:
```
@layer base {
  .cards--mobile-none {
    @apply -mx-4 md:mx-0;
  }
}
```

**Comment by wongjn** [comment]
> those css classes are set in CMS systems, so they never exist for tailwind content watch; but since classes are set in .css they should always be generated?

No, your classes wouldn't be generated, as per [the documentation](https://tailwindcss.com/docs/adding-custom-styles#using-css-and-layer) (emphasis mine):

> The `@layer` directive helps you control declaration order by automatically relocating your styles to the corresponding `@tailwind` directive, and also enables features like [modifiers](https://tailwindcss.com/docs/adding-custom-styles#using-modifiers-with-custom-css) and [*tree-shaking*](https://tailwindcss.com/docs/adding-custom-styles#removing-unused-custom-css) for your own custom CSS.

Since the rules are inside `@layer base` they become "managed" by Tailwind. Thus they act like any other class name in Tailwind. If Tailwind does not see the class name as a string in `content`, it will remove it.

If you need to rules to always be compiled, consider unwrapping them from the `@layer base`.
